src/core/core.controller.js

```diff
@@ -784,7 +796,7 @@ class Chart {
     const ctx = this.ctx;
     const clip = meta._clip;
     const useClip = !clip.disabled;
-    const area = this.chartArea;
+    const area = getDatasetArea(meta) || this.chartArea;
```

Comment:
Is the fallback to chartArea still needed?

Reply:
I think yes because it is invoked also from controllers without scales or with a radial one, where the clip should be the chart area. But correct me if I'm wrong.